GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,68 @@
+﻿namespace GenericsHomework
+{
+    public class Node<TValue>
+        where TValue : notnull
+    {
+        private TValue Value { get; set; }
+        public Node<TValue> Next { get; private set; }
+
+        public Node(TValue value)
+        {
+            Value = value;
+            Next = this;
+        }
+
+        public void Clear()
+        {
+            this.Next = this;
```

Comment:
Great! You should include a comment as to why this works though.
```suggestion
            Next = this;
```